src/cmd-buildextend-gcp

```diff
@@ -0,0 +1,133 @@
+                    default=bool(os.environ.get("GCP_GCE_PLATFORM_ID", False)))
+parser.add_argument("--force", help="Replace existing images and upload",
+                    action="store_true",
+                    default=os.environ.get("GCP_FORCE", False))
+parser.add_argument("--json-key", help="GCP Service Account JSON Auth",
+                    default=os.environ.get("GCP_JSON_AUTH", None))
+parser.add_argument("--name-suffix", help="Append suffix to name",
+                    required=False)
+parser.add_argument("--project", help="GCP Project name",
+                    default=os.environ.get("GCP_PROJECT_NAME", None))
+args = parser.parse_args()
+
+# Argument checks for environment strings that are required
+arg_exp_str = "parameter '--{}' or envVar '{}' must be defined"
+if args.bucket is None:
+    raise Exception(arg_exp_str.format("bucket", "GCP_BUCKET"))
+if args.json_key is None:
+    raise Exception(arg_exp_str.format("json-key", "GCP_JSON_AUTH"))
+if args.project is None:
+    raise Exception(arg_exp_str.format("project", "GCP_PROJECT"))
+
+# Identify the builds
+builddir = os.path.join('builds', args.build)
+buildmeta_path = os.path.join(builddir, 'meta.json')
+with open(buildmeta_path) as f:
+    buildmeta = json.load(f)
+
+if 'gcp' in buildmeta['images'] and not args.force:
+  print("GCP image already built; use --force to rebuild")
+  sys.exit(1)
+
+# Name the base build and tarball name
+base_name = buildmeta['name']
+if args.name_suffix:
+    gcp_name_version = f'{base_name}-{args.name_suffix}-{args.build}'
+else:
+    gcp_name_version = f'{base_name}-{args.build}'
+gcp_tarball_name = f'{gcp_name_version}-gcp.tar.gz'
+
+# Setup the tempdir
+tmpdir = 'tmp/buildpost-gcp'
+if os.path.isdir(tmpdir):
+    shutil.rmtree(tmpdir)
+os.mkdir(tmpdir)
+tmpdir = os.path.abspath(tmpdir)
+
+def generate_gcp_tar():
+    """ Set the Ignition ID to GCP/GCE and create a tarball for upload """
+    plat_id = 'gcp'
+    if args.gce:
+        plat_id = 'gce'
+    img_qemu = os.path.join(builddir, buildmeta['images']['qemu']['path'])
+    tmp_img_gcp = os.path.join(tmpdir, (gcp_name_version + '.qcow2'))
+    tmp_img_gcp_raw = os.path.join(tmpdir, 'disk.raw')
+    tmp_img_gcp_tar = os.path.join(tmpdir, gcp_tarball_name)
+    run_verbose([f"{cosa_dir}/gf-platformid", img_qemu, tmp_img_gcp, plat_id])
+    run_verbose(['qemu-img', 'convert', '-f', 'qcow2', '-O', 'raw',
+                 tmp_img_gcp, tmp_img_gcp_raw])
+    run_verbose(['tar', '-C', f"{tmpdir}", '-Sczf', f"{tmp_img_gcp_tar}",
+                 f"disk.raw"])
+    os.unlink(tmp_img_gcp)
+    os.unlink(tmp_img_gcp_raw)
+    return tmp_img_gcp_tar
+
+def run_ore():
+    """ Execute ore to upload the tarball and register the image """
+    build_tarball = f"{builddir}/{gcp_tarball_name}"
+    tmp_img_gcp_tarball = generate_gcp_tar()
+    # remove the user from the environ, otherwise ore will put the user's name
+    # as a prefix to the upload.
+    del os.environ['USER']
```

Comment:
Hmm, is that actually required? In `ore/gcloud/upload.go` I see:

```go
	gsURL, err := url.Parse(uploadBucket)
	if err != nil {
		fmt.Fprintf(os.Stderr, "%v\n", err)
		os.Exit(1)
	}
	if gsURL.Scheme != "gs" {
		fmt.Fprintf(os.Stderr, "URL missing gs:// scheme prefix: %v\n", uploadBucket)
		os.Exit(1)
	}
	if gsURL.Host == "" {
		fmt.Fprintf(os.Stderr, "URL missing bucket name %v\n", uploadBucket)
		os.Exit(1)
	}
	// if prefix not specified default name to gs://bucket/$USER/$BOARD/$VERSION
	if gsURL.Path == "" {
		if user := os.Getenv("USER"); user != "" {
			gsURL.Path = "/" + os.Getenv("USER")
			gsURL.Path += "/" + uploadBoard
		}
	}

```

And here `gsURL.Path` should be `{base_name}`, right?

Reply:
Indeed, you are correct. Removed.